nativeApp/src/lib/ui/forms/UserForm/index.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, useState, useEffect } from 'react';
+import styled from 'styled-components/native';
+import { Form as F } from 'native-base';
+
+import Input from '../InputWithInlineLabel';
+import Button from '../ButtonWithInlineLabel';
+import Dropdown from '../Dropdown';
+import Toggle from '../Toggle';
+import TermsAndConditonsButton from './TermsAndConditionsButton';
+import { Forms } from '../enums';
+import SubmitButton from '../SubmitButton';
+
+/*
+ * Types
+ */
+type UserValues = {
+  name: string;
+  email: string;
+  number: string;
+  postcode: string;
+  gender: string;
+  year: string;
+}
+
+type Props = {
+  onSubmit: (v: UserValues) => void;
+  initialValues?: UserValues;
+}
+
+/*
+ * Styles
+ */
+const Form = styled(F)`
+  width: ${Forms.formWidth}
+`;
+
+/*
+ * Component
+ */
+const genders = [
+  { id: 1, name: 'female' },
+  { id: 2, name: 'male' },
+  { id: 3, name: 'other' },
+];
+
+const years = [...Array(130).keys()].map((_, i) => ({ id: i, name: `${2019 - i}` }));
+
+const UserForm: FC<Props> = ({ onSubmit, initialValues = {} }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [number, setNumber] = useState('');
+  const [postcode, setPostcode] = useState('');
+
+  const [gender, setGender] = useState('');
+  const [year, setYear] = useState('');
+
+  const useSubmit = () => {
```

Comment:
How come this is a `useEffect` wrapped in a callback instead of just a `useCallback`?

Reply:
I forgot `useCallback` existed 😅